FreeBSD building error

```
[ 95%] Building CXX object src/daemon/CMakeFiles/daemon.dir/rpc_command_executor.cpp.o
[ 95%] Linking CXX executable ../../bin/monerod
/usr/bin/ld: /usr/local/lib/libminiupnpc.a(miniupnpc.c.o): relocation R_X86_64_32S against `IGDstartelt' can not be used when making a shared object; recompile with -fPIC
/usr/local/lib/libminiupnpc.a: could not read symbols: Bad value
c++: error: linker command failed with exit code 1 (use -v to see invocation)
```
any ideas?

pull last source and get this error:
```
[ 96%] Building CXX object src/blockchain_utilities/CMakeFiles/blockchain_blackball.dir/blockchain_blackball.cpp.o
/monero/src/blockchain_utilities/blockchain_blackball.cpp:173:9: error: return type 'basic_string<[3 * ...]>' must match previous return
      type 'const basic_string<[3 * ...]>' when lambda expression has unspecified explicit return type
        return val;
        ^
In file included from /monero/src/blockchain_utilities/blockchain_blackball.cpp:33:
In file included from /monero/src/cryptonote_core/tx_pool.h:44:
In file included from /monero/src/cryptonote_basic/cryptonote_basic_impl.h:33:
In file included from /monero/src/cryptonote_basic/cryptonote_basic.h:41:
/monero/src/serialization/binary_archive.h:194:28: warning: shift count >= width of type [-Wshift-count-overflow]
      if (1 < sizeof(T)) v >>= 8;
                           ^   ~
/monero/src/serialization/binary_archive.h:187:5: note: in instantiation of function template specialization
      'binary_archive<true>::serialize_uint<unsigned char>' requested here
    serialize_uint(static_cast<typename boost::make_unsigned<T>::type>(v));
    ^
/monero/src/serialization/binary_archive.h:227:5: note: in instantiation of function template specialization
      'binary_archive<true>::serialize_int<unsigned char>' requested here
    serialize_int(t);
    ^
1 warning and 1 error generated.
*** Error code 1

Stop.
make[2]: stopped in /monero
*** Error code 1

Stop.
make[1]: stopped in /monero
*** Error code 1

Stop.
make: stopped in /monero
```

--- a/src/blockchain_utilities/blockchain_blackball.cpp
+++ b/src/blockchain_utilities/blockchain_blackball.cpp
@@ -165,7 +165,7 @@ int main(int argc, char* argv[])
       "blackball-db-dir", "Specify blackball database directory",
       get_default_db_path(),
       {{ &arg_testnet_on, &arg_stagenet_on }},
-      [](std::array<bool, 2> testnet_stagenet, bool defaulted, std::string val) {
+      [](std::array<bool, 2> testnet_stagenet, bool defaulted, std::string val)->std::string {
         if (testnet_stagenet[0])
           return (boost::filesystem::path(val) / "testnet").string();
         else if (testnet_stagenet[1])

